### PR TITLE
Only run full profile on Jenkins

### PR DESCRIPTION
**Pull request by triceo**

### CODE REVIEW

File: .github/workflows/os_java_ci.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ jobs:
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-m2
       - name: Build with Maven
-        run: mvn -B --fail-at-end clean install -Dfull
+        run: mvn -B --fail-at-end clean install
 
   # Early feedback on a compatibility with the newest Java version.
   # TODO: update each time a new major Java version is released.
```
Comment (rsynek) [inline .github/workflows/os_java_ci.yml:43]:
```suggestion
  # TODO: Update each time a new major Java version is released. Make sure you build OptaPlanner with the full profile locally with the new Java version first.
```

**Review by rsynek — APPROVED** [review]
One suggestion to increase the chance we won't forget to test the full profile with the new JDK.

**Review by ge0ffrey — CHANGES_REQUESTED** [review]
I 'd rather stop relying on Jenkins. It's not reliable.
Many important checks only happen in the full profile.

**Comment by triceo** [comment]
There is no consensus on this.

**Pull request closed without merging**